src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -167,17 +167,18 @@ private ClassLoader getScannerClassLoader() throws MalformedURLException {
         return binaryScannerClassLoader;
     }
 
-    private static String[] getBinaryInputs(List<String> classFiles, Set<String> classDirectories) throws PluginExecutionException {
+    private static String[] getBinaryInputs(List<String> classFiles, Set<String> classDirectories, boolean optimize) throws PluginExecutionException {
         Collection<String> resultSet;
-        if (classFiles != null && !classFiles.isEmpty()) {
+        if (!optimize && classFiles != null && !classFiles.isEmpty()) {
             resultSet = classFiles;
-        } else {
+        } else if (optimize) {
             if (classDirectories == null || classDirectories.isEmpty()) {
                 throw new PluginExecutionException("Error collecting list of directories to send to binary scanner, list is null or empty.");
             }
             resultSet = classDirectories;
+        } else {
+            return new String[0];
```

Comment:
This set of if-statements is confusing because they have multiple conditions and it's not immediately obvious what are the conditions for the last `else` .  Seems that is for !optimize and classFiles are null or empty.  In this case, consider rearranging the conditions to the following for better readability:
```
if (optimize) {
  ...
} else {
  if (classFiles != null && !classFiles.isEmpty())) {
    ...
  } else {
    ...
  }
}

Reply:
I will rearrange the logic here.